forc-plugins/forc-client/src/op/deploy.rs

```diff
@@ -75,6 +148,7 @@ pub async fn deploy_pkg(
     command: &cmd::Deploy,
     manifest: &PackageManifestFile,
     compiled: &BuiltPackage,
+    contract_salt_map: Option<&ContractSaltMap>,
```

Comment:
Rather than passing in the whole `ContractSaltMap` here, perhaps we should determine the salt before calling `deploy_pkg` and pass the package's `Salt` as an arg?